jwst/tweakreg/tweakreg_step.py

```diff
@@ -300,6 +312,28 @@ def process(self, input):
 
         return images
 
+    def fit_quality_is_good(self, imcat):
+        """Check that the newly tweaked wcs hasn't gone off the rails"""
+        tolerance = 10.0 * u.arcsec
+
+        wcs = imcat.meta['image_model'].meta.wcs
+        twcs = imcat.wcs
+
+        grid = grid_from_bounding_box(wcs.bounding_box, step=100)
+        # There's a bug in grid_from_bounding_box that makes the final grid
+        # item outside the bounding box.  So trim off the last one.  And trim
+        # the first one to avoid edge effects.
+        grid = grid[:,1:-1,1:-1]
```

Comment:
I think this is an overkill. Using the corners of the bounding box should be enough: these are extreme points of the image and should reveal large shift, scale, and/or rotation.

Reply:
Great feedback.  `wcs.footprint().T` gives just the bounding box corners in RA/Dec, so I'll just use those.